File: backend/Perflow/Common/DTO/User/UserDTO.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Perflow.Common.DTO.User
{
    public class UserDTO
    {
        public int Id { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
        public string UserName { get; set; }
        public string IconURL { get; set; }
        public string Email { get; set; }
        public DateTimeOffset Birthday { get; set; }
        public bool Gender { get; set; }
        public string Description { get; set; }
        public string Country { get; set; }
        public string Password { get; set; }
        public string Salt { get; set; }
        public int? GroupId { get; set; }
        public ICollection<ArtistFollowerDTO> Subscriptions { get; set; }
    }
}

File: frontend/src/app/models/user/UserDTO.ts
export class UserDTO {
  id!: number;
  createdAt: Date;
  userName!: string;
  iconURL!: string;
  email!: string;
  birthday!: Date;
  gender!: boolean;
  description!: string;
  country!: string;
  password!: string;
  salt!: string;
  groupId: number;
}
